src/components/List/index.tsx

```diff
@@ -108,21 +136,29 @@ const List = ({ items }: Props): React.ReactElement => {
   const classes = useStyles()
   const [groupCollapseStatus, setGroupCollapseStatus] = useState({})
 
-  const onItemClick = (item: ListItemType) => {
-    if (item.subItems) {
-      const cp = { ...groupCollapseStatus }
-      cp[item.label] = cp[item.label] ? false : true
-      setGroupCollapseStatus(cp)
+  const onItemClick = (item: ListItemType, isSubItem: boolean, event: MouseEvent) => {
+    const collapseStatus = { ...groupCollapseStatus }
+
+    // In the current implementation we only want to allow one expanded item at a time
+    // When we click any entry that is not a subItem we want to collapse all current expanded items
+    if (!isSubItem && !item.selected) {
+      const collapseKeys = Object.keys(collapseStatus)
+      collapseKeys.forEach((key) => (collapseStatus[key] = false))
+      setGroupCollapseStatus(collapseStatus)
```

Comment:
Nice improvement. But then I also need to change the next if approach to use this updated state :+1:

Reply:
Although I'm thinking if resetting to `{}` would be a better improvement, because it will need less operations and we may end having a big object if jumping between many safes and using `item.href` instead of `item.label` as key.

That made me also detect inconsistent behaviour when switching between safes :wink: